packages/teleport-generator-core/src/validator/index.ts

```diff
@@ -38,6 +42,66 @@ export default class Validator {
 
     return { valid: true, errorMsg: '' }
   }
+
+  public validateComponentContent(input: any): ValidationResult {
+    const verifyProps = checkContent(input, 'prop')
+    const verifyState = checkContent(input, 'state')
+
+    checkForDuplicates(input)
+    const errors = verifyProps.errors.concat(verifyState.errors)
+
+    if (errors.length > 0) {
+      return {
+        valid: false,
+        errorMsg: `\nUIDL Content Validation Error. Please check the following: \n${errors}`,
+      }
+    }
+
+    return { valid: true, errorMsg: '' }
+  }
+}
+
+const checkForDuplicates = (input: any) => {
+  const props = Object.keys(input.propDefinitions || [])
+  const states = Object.keys(input.stateDefinitions || [])
+
+  const duplicates = props.filter((x) => states.includes(x))
+
+  if (duplicates.length > 0) {
+    duplicates.map((duplicate) =>
+      console.warn(
+        `"${duplicate}" is defined both as a prop and as a state. If you are using VUE Code Generators this can cause bad behavior.`
+      )
+    )
+  }
+}
+
+const checkContent = (input: any, type: string) => {
+  const definitions = `${type}Definitions`
+  const definedKeys = Object.keys(input[definitions] || [])
+  const usedKeys = []
+  const errors = []
+
+  traverseNodes(input.node, (node) => {
+    if (node.type === 'dynamic' && node.content.referenceType === type) {
+      if (!definedKeys.includes(node.content.id)) {
+        const errorMsg = `"${
+          node.content.id
+        }" is used as ${type} but not defined. Please add it in ${definitions}`
+        errors.push(errorMsg)
+      }
+      usedKeys.push(node.content.id)
+    }
+  })
+
+  const diffs = definedKeys.filter((x) => !usedKeys.includes(x))
```

Comment:
you can pipe the `filter` and the `map` as the map will be skipped in case no values are left after the filter